app/api/csv/arrangeThesauri.ts

```diff
@@ -0,0 +1,180 @@
+
+  return { thesauriIdToExistingValues, thesauriIdToNewValues, thesauriIdToNormalizedNewValues };
+};
+
+const handleLabels = (
+  id: string,
+  original: string,
+  normalized: string | null,
+  thesauriValueData: ThesauriValueData
+) => {
+  if (
+    normalized &&
+    !thesauriValueData.thesauriIdToExistingValues.get(id)?.has(normalized) &&
+    !thesauriValueData.thesauriIdToNormalizedNewValues.get(id)?.has(normalized)
+  ) {
+    thesauriValueData.thesauriIdToNewValues.get(id)?.add(original);
+    thesauriValueData.thesauriIdToNormalizedNewValues.get(id)?.add(normalized);
+  }
+};
+
+const syncSaveThesauri = async (
+  allRelatedThesauri: ThesaurusSchema[],
+  thesauriIdToNewValues: Map<string, Set<string>>
+) => {
+  for (let i = 0; i < allRelatedThesauri.length; i += 1) {
+    const thesaurus = allRelatedThesauri[i];
+    if (thesaurus?._id) {
+      const newValues: { label: string }[] = Array.from(
+        thesauriIdToNewValues.get(thesaurus._id.toString()) || []
+      ).map(tval => ({ label: tval }));
+      if (newValues.length > 0) {
+        const thesaurusValues = thesaurus.values || [];
+        // eslint-disable-next-line no-await-in-loop
+        await thesauri.save({
+          ...thesaurus,
+          values: thesaurusValues.concat(newValues),
+        });
+      }
+    }
+  }
+};
+
+const arrangeThesauri = async (
+  file: ImportFile,
+  template: TemplateSchema,
+  languages?: string[],
+  stopOnError: boolean = true
+) => {
+  const thesauriRelatedProperties = template.properties?.filter(p =>
+    ['select', 'multiselect'].includes(p.type)
+  );
+
+  let [
+    nameToThesauriIdSelects,
+    nameToThesauriIdMultiselects,
+  ] = separateSelectAndMultiselectThesauri(thesauriRelatedProperties, languages);
+
+  const allRelatedThesauri = await thesauri.get({
+    $in: Array.from(
+      new Set(thesauriRelatedProperties?.map(p => p.content?.toString()).filter(t => t))
+    ),
+  });
+
+  const thesauriValueData = setupIdValueMaps(allRelatedThesauri);
+
+  await csv(await file.readStream(), stopOnError)
+    .onRow(async (row: CSVRow, index: number) => {
+      if (index === 0) {
+        const columnnames = Object.keys(row);
+        nameToThesauriIdSelects = filterJSObject(nameToThesauriIdSelects, columnnames);
+        nameToThesauriIdMultiselects = filterJSObject(nameToThesauriIdMultiselects, columnnames);
+      }
+      Object.entries(nameToThesauriIdSelects).forEach(([name, id]) => {
+        const label = row[name];
+        if (label) {
+          const normalizedLabel = normalizeThesaurusLabel(label);
+          handleLabels(id, label, normalizedLabel, thesauriValueData);
+        }
+      });
+      Object.entries(nameToThesauriIdMultiselects).forEach(([name, id]) => {
+        const labels = splitMultiselectLabels(row[name]);
+        if (labels) {
+          Object.entries(labels).forEach(([normalizedLabel, originalLabel]) => {
+            handleLabels(id, originalLabel, normalizedLabel, thesauriValueData);
```

Comment:
`handleLables` its not clear what it does, i think something like `addNewLabel` is more descriptive of what the function does ?
also this function modifies a passed parameter and it makes it dificult to follow, consider modifying the valueData in here.

Reply:
Since we do not separate selects and multiselects, this thing is used only once, so the separate function is removed and the modification is done here.